src/components/ListingDetails/index.tsx

```diff
@@ -0,0 +1,323 @@
+          {
+            label: 'Client Location',
+            getValue: (data) =>
+              data.client_location || 'N/A',
+          },
+        ];
+      }
+    } else if (listingData.listing_type === 'INT') {
+      fields = intepretationFields;
+    } else if (listingData.listing_type === 'DOC') {
+      fields = docFields;
+    } else {
+      fields = lcaFields;
+    }
+
+    return (
+      <InnerFieldContainer>
+        {fields.map(({ label, getValue }) => (
+          <div key={label}>{renderField(label, getValue(listingData))}</div>
+        ))}
+      </InnerFieldContainer>
+    );
+  }, [listingData, interpretation]);
+
+  const dateComponent = () => {
+    if (listingData.listing_type !== 'INT') {
+      let dateHeader = 'Assignment Deadline';
+      let dateData = '';
+      if (listingData.listing_type === 'CASE') {
+        dateHeader = 'Next Court/Filing Date';
+        dateData = formatTimestamp(listingData.upcoming_date); // listingData.upcoming_date ? formatTimestamp(listingData.upcoming_date) : "N/A"
+      } else {
+        dateData = formatTimestamp(listingData.deadline);
+      }
+      return (
+        <IconTextGroup>
+          <Icon type="calendar" />
+          <DateText $bold>
+            {dateHeader}: <DateText> {dateData}</DateText>
+          </DateText>
+        </IconTextGroup>
+      );
+    }
+    return null;
+  };
+
+  const profile = useContext(ProfileContext);
+
+  const Interest = useMemo(() => {
+    if (profile && profile.userId)
+      return profile.profileData ? (
+        <InterestForm caseData={listingData as CaseListing} />
+      ) : (
+        <>
+          <H3>
+            Please finish submitting your profile before submitting interest.
+          </H3>
+          <AuthButtons>
+            <LinkButton
+              $primaryColor={COLORS.blueMid}
+              $secondaryColor={COLORS.blueDark}
+              href="/onboarding/roles"
+            >
+              Go to Onboarding
+            </LinkButton>
+          </AuthButtons>
+        </>
+      );
+
+    return (
+      <>
+        <H3>
+          You aren&apos;t signed in. In order to submit your interest, log into
+          your account, or create a new account.
+        </H3>
+        <AuthButtons>
+          <LinkButton $secondaryColor={COLORS.blueMid} href="/signup">
+            Sign Up
+          </LinkButton>
+
+          <LinkButton
+            $primaryColor={COLORS.blueMid}
+            $secondaryColor={COLORS.blueDark}
+            href="/login"
+          >
+            Log In
+          </LinkButton>
+        </AuthButtons>
+      </>
+    );
+  }, [listingData, profile]);
+
+  return (
+    <CaseDisplay>
+      <InfoContainer>
+        <BorderedSection>
+          {dateComponent()}
+          <Flex $align="center" $gap="18px" $direction="row">
+            <img src={logo.src} alt="DOHS Logo" width="77" height="77" />
+            <H2>{listingData.title || 'Migrant seeking representation'}</H2>
```

Comment:
can you refactor this to be different depending on the type of listing? (i.e. defaults for LCA and the various language support types)

we can ask Ambreen for the specifics on Monday, but feel free to add arbitrary defaults for now!

Reply:
i'll leave it as is for now, since all listing types besides cases have non-nullable titles, so hopefully the a default title will never be needed for non-case listings.